File: pubnub-gson/pubnub-gson-docs/src/main/java/com/pubnub/docs/appContext/ChannelMetadataOthers.java
package com.pubnub.docs.appContext;

import com.pubnub.api.PubNubException;
import com.pubnub.api.java.PubNub;
import com.pubnub.api.java.models.consumer.objects_api.channel.PNGetAllChannelsMetadataResult;
import com.pubnub.api.java.models.consumer.objects_api.channel.PNGetChannelMetadataResult;
import com.pubnub.docs.SnippetBase;

public class ChannelMetadataOthers extends SnippetBase {
    private void getAllChannelsMetadataBasic() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/objects#basic-usage-4

        PubNub pubNub = createPubNub();

        // snippet.getAllChannelsMetadataBasic
        PNGetAllChannelsMetadataResult pnGetAllChannelsMetadataResult = pubNub.getAllChannelsMetadata().sync();
        // snippet.end
    }

    private void getChannelMetadataBasic() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/objects#basic-usage-5

        PubNub pubNub = createPubNub();

        // snippet.getChannelMetadataBasic
        PNGetChannelMetadataResult pnGetChannelMetadataResult = pubNub.getChannelMetadata()
                .channel("myChannel")
                .sync();
        // snippet.end

    }

    private void setChannelMetadataBasic() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/objects#basic-usage-6

        PubNub pubNub = createPubNub();

        // snippet.setChannelMetadataBasic
        pubNub.setChannelMetadata()
                .channel("myChannel")
                .name("Some Name")
                .includeCustom(true)
                .async(result -> { /* check result */ });
        // snippet.end
    }

    private void removeChannelMetadataBasic() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/objects#basic-usage-7

        PubNub pubNub = createPubNub();

        // snippet.removeChannelMetadataBasic
        pubNub.removeChannelMetadata()
                .channel("myChannel")
                .async(result -> { /* check result */ });
        // snippet.end
    }
}

File: pubnub-kotlin/pubnub-kotlin-docs/src/main/kotlin/com/pubnub/docs/appContext/ChannelMetadataOthers.kt
package com.pubnub.docs.appContext

import com.pubnub.api.PubNub

class ChannelMetadataOthers {
    private fun getAllChannelMetadataBasic(pubnub: PubNub) {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/objects#basic-usage-4

        // snippet.getAllChannelMetadataBasic
        pubnub.getAllChannelMetadata()
            .async { result ->
                result.onFailure { exception ->
                    // Handle error
                }.onSuccess { value ->
                    // Handle successful method result
                }
            }
        // snippet.end
    }

    private fun getChannelMetadataBasic(pubnub: PubNub) {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/objects#basic-usage-5

        // snippet.getChannelMetadataBasic
        pubnub.getChannelMetadata(channel = "myChannel")
            .async { result ->
                result.onFailure { exception ->
                    // Handle error
                }.onSuccess { value ->
                    // Handle successful method result
                }
            }
        // snippet.end
    }

    private fun setChannelMetadataBasic(pubnub: PubNub) {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/objects#basic-usage-6

        // snippet.setChannelMetadataBasic
        pubnub.setChannelMetadata(channel = "myChannel")
            .async { result ->
                result.onFailure { exception ->
                    // Handle error
                }.onSuccess { value ->
                    // Handle successful method result
                }
            }
        // snippet.end
    }

    private fun removeChannelMetadata(pubnub: PubNub) {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/objects#basic-usage-7

        // snippet.removeChannelMetadata
        pubnub.removeChannelMetadata(channel = "myChannel")
            .async { result ->
                result.onFailure { exception ->
                    // Handle error
                }.onSuccess { value ->
                    // Handle successful method result
                }
            }
        // snippet.end
    }
}
